android/src/main/java/com/op/sqlite/OPSQLiteModule.kt

```diff
@@ -47,25 +50,33 @@ internal class OPSQLiteModule(context: ReactApplicationContext?) : ReactContextB
         }
     }
 
-    @ReactMethod(isBlockingSynchronousMethod = true)
-    fun moveAssetsDatabase(name: String, extension: String): Boolean {
+    @ReactMethod
+    fun moveAssetsDatabase(args: ReadableMap, promise: Promise) {
+        val filename = args.getString("filename")!!
+        val path = args.getString("path") ?: "custom"
+        val overwrite = if(args.hasKey("overwrite")) { args.getBoolean("overwrite") } else false
         val context = reactApplicationContext
         val assetsManager = context.assets
 
         try {
             // Open the input stream for the asset file
-            val inputStream: InputStream = assetsManager.open("custom/$name.$extension")
+            val inputStream: InputStream = assetsManager.open("$path/$filename")
 
             // Create the output file in the documents directory
             val databasesFolder =
                     context.getDatabasePath("defaultDatabase")
                             .absolutePath
                             .replace("defaultDatabase", "")
 
-            val outputFile = File(databasesFolder, "$name.$extension")
+            val outputFile = File(databasesFolder, filename)
 
             if (outputFile.exists()) {
-                return true
+                if(overwrite) {
+                    outputFile.delete()
+                } else {
+                    promise.resolve(true)
```

Comment:
Shouldn't the input stream be closed here before resolving the promise and returning?

Reply:
Solved by moving the instantiation down